src/NodaTime.Web/Startup.cs

```diff
@@ -80,6 +84,43 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                 ContentTypeProvider = new FileExtensionContentTypeProvider
                 {
                     Mappings = { [".nzd"] = "application/octet-stream" }
+                },
+                OnPrepareResponse = (context) => {
+                    // Set the Cache-Control header for static content.
+                    var headers = new ResponseHeaders(context.Context.Response.Headers);
+
+                    // All static content is cacheable by intermediates.
+                    headers.CacheControl = new CacheControlHeaderValue()
+                    {
+                        Public = true
+                    };
+
+                    if (headers.ContentType.IsSubsetOf(TextHtml))
+                    {
+                        // Don't set max-age for HTML files (e.g. /tzdb/). We'll just send an ETag and let the browser
+                        // figure out when to revalidate.
+                        return;
+                    }
+
+                    // Otherwise send a sensible max-age value. If the request was made with a file version query
+                    // parameter (?v=..., as sent by asp-append-version=true), then cache the response 'indefinitely'.
+                    // Otherwise (e.g. /favicon.ico, /fonts/, /robots.txt), cache for a day.
+                    if (context.Context.Request.Query.ContainsKey("v"))
+                    {
+                        headers.CacheControl = new CacheControlHeaderValue()
```

Comment:
I can write `headers.CacheControl.MaxAge = ...` here, but while it compiles, it doesn't do anything.

I can only assume that the getters for the `ResponseHeaders` type return mutable value-type objects, rather than proxies, and so it's actually necessary to use the setters to get anything to propagate through to the underlying `Context.Response.Headers` dictionary.